src/main/java/junmin/netflix/costreportservice/service/EPCostReportService.java

```diff
@@ -0,0 +1,169 @@
+package junmin.netflix.costreportservice.service;
+
+import junmin.netflix.costreportservice.exception.ErrorEnum;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+//import junmin.netflix.costreportservice.model.EPItemizedCostEntity;
+import junmin.netflix.costreportservice.model.EPTotalCostEntity;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.repository.EPItemizedCostRepository;
+import junmin.netflix.costreportservice.repository.EPTotalCostRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class EPCostReportService {
+	private EPTotalCostRepository epTotalCostRepo;
+	private EPItemizedCostRepository epItemizedCostRepo;
+
+	public EPCostReportService(EPTotalCostRepository epTotalCostRepo, EPItemizedCostRepository epItemizedCostRepo){
+		this.epTotalCostRepo = epTotalCostRepo;
+		this.epItemizedCostRepo = epItemizedCostRepo;
+	}
+
+	private final Logger LOGGER = LoggerFactory.getLogger(EPCostReportService.class);
+	private final String AMORIZEDCODE = "001";
```

Comment:
Any code beginning with 0 is an amortized code.

Reply:
You are right. I realized this when I re-read the requirement document again last night, it said if it's season one, then the amortizedCode is 001. I did not really implement the logic correctly in case we have multiple amortized costs cross multiple seasons. this is a bug due to mis-interpret the requirements and need to be fixed.